zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/tasks/incident/IncidentUpdateTask.java

```diff
@@ -313,27 +304,31 @@ private void createFlowNodeInstanceUpdates(
 
   private void createFlowNodeInstanceUpdate(
       final AdditionalData data,
-      final String incidentId,
+      final IncidentDocument incident,
       final IncidentState newState,
       final String fniId,
       final List<String> flowNodeIndices,
       final IncidentBulkUpdate updates,
-      final List<String> listViewIndices,
-      final String routing) {
+      final List<String> listViewIndices) {
     final var hasIncident = IncidentState.ACTIVE == newState;
     final boolean changedState;
     if (hasIncident) {
-      changedState = data.addFniIdsWithIncidentIds(fniId, incidentId);
+      changedState = data.addFniIdsWithIncidentIds(fniId, incident.id());
     } else {
-      changedState = data.removeIncidentIdByFniId(fniId, incidentId);
+      changedState = data.removeIncidentIdByFniId(fniId, incident.id());
     }
 
+    final var treePath = new TreePath(incident.incident().getTreePath());
```

Comment:
:question: why making a copy of the TreePath if only need to extract the process instance id? Is it because it is mutable?

Reply:
It's not a copy, it's wrapping the String to make it a `TreePath` type. Or do you mean why is `TreePath` copying it to `StringBuffer`? I guess because it's mutable, yeah.

Reply:
Ah sorry I thought it was a copy-constructor, but `getTreePath` is just a string. :+1: